shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -399,7 +399,7 @@ private ViewAndId(View view, int id) {
     @Override
     public boolean equals(Object o) {
       if (this == o) return true;
-      if (o == null || getClass() != o.getClass()) return false;
+      if (!(o instanceof ViewAndId)) return false;
```

Comment:
avoid class comparison in equals in non-final class

Reply:
This one looks like it might have had a reason for being written that way, but since it is private and not subclassed that hypothetical reason wouldn't apply.

(Basically an "ID" type of class could potentially have the contract that it only matches if the exact class matches, whatever subclass that may be. Consider if this was subclassed by "FooViewId" and "BarViewId" and those were not supposed to match each other. You could put the burden on the subclasses to each provide an appropriate "equals()" for their specific subtype, but those lines of code could enforce that contract for them. In either case, there would have to be subclasses of ViewAndId for that to even be a "thing"...)

Reply:
If that's the case, it should be final right?

We could alternatively make the class final.

Reply:
(as in, the class should be final)

Reply:
This particular class has no subclasses so it could be final, but I'm not sure it matters. I was just speculating that in a different situation an "ID" class might have a use for that paradigm.

The method could be final, but subclasses may have additional checks to make and so may want to have { if (super.equals(other) && myStuff == other.myStuff)... }.

I'm not suggesting any changes here, I'm just pointing out that the lint rule being followed isn't necessarily an "always" thing...